.meta/sources/host_metrics.toml.erb

```diff
@@ -0,0 +1,94 @@
+[sources.host_metrics]
+title = "Host Metrics"
+noun = "Host Metrics"
+beta = true
+common = false
+delivery_guarantee = "at_least_once"
+features = [
+  "Gather host-based metrics.",
+]
+function_category = "collect"
+output_types = ["metric"]
+requirements = {}
+strategies = ["daemon", "sidecar", "service"]
+through_description = "local system data sources"
+
+<%= render("_partials/fields/_component_options.toml", type: "source", name: "host_metrics") %>
+
+[sources.host_metrics.options.collectors]
+type = "[string]"
+common = false
+required = false
+description = "The list of host metric collector services to use. Defaults to all collectors."
+
+[sources.host_metrics.options.collectors.enum]
+cpu = "Metrics related to CPU utilization."
+disk = "Metrics related to disk I/O utilization."
+filesystem = "Metrics related to filesystem space utilization."
+load = "Load average metrics (UNIX only)."
+memory = "Metrics related to memory utilization."
+network = "Metrics related to network utilization."
+
+[sources.host_metrics.options.namespace]
+type = "string"
+common = false
+required = false
+default = "host"
+description = "The namespace prefix that will be added to all metric names."
+
+[sources.host_metrics.options.scrape_interval_secs]
+type = "integer"
+common = false
+required = false
+default = 15
+description = "The interval between metric gathering, in seconds."
+
+[sources.host_metrics.options.disk]
+type = "table"
+category = "Filesystem"
+description = "Options for the \"disk\" metrics collector."
+
+<%= render("_partials/fields/_includes_excludes.toml",
+  namespace: "sources.host_metrics.options.disk.options.devices",
+  term: "device name",
+  metrics: "I/O",
+  examples: [["sda"], ["dm-*"]]
+) %>
+
+[sources.host_metrics.options.filesystem]
+type = "table"
+category = "Filesystem"
+description = "Options for the \"filesystem\" metrics collector."
+
+<%= render("_partials/fields/_includes_excludes.toml",
+  namespace: "sources.host_metrics.options.filesystem.options.devices",
+  term: "device name",
+  metrics: "usage",
+  examples: [["sda"], ["dm-*"]]
+) %>
+
+<%= render("_partials/fields/_includes_excludes.toml",
+  namespace: "sources.host_metrics.options.filesystem.options.filesystems",
+  term: "filesystem name",
+  metrics: "usage",
+  examples: [["ntfs"], ["ext*"]]
+) %>
+
+<%= render("_partials/fields/_includes_excludes.toml",
+  namespace: "sources.host_metrics.options.filesystem.options.mountpoints",
+  term: "mountpoint path",
+  metrics: "usage",
+  examples: [["/home"], ["/raid*"]]
+) %>
+
+[sources.host_metrics.options.network]
+type = "table"
+category = "Network"
+description = "Options for the \"network\" metrics collector."
+
+<%= render("_partials/fields/_includes_excludes.toml",
+  namespace: "sources.host_metrics.options.network.options.devices",
+  term: "mountpoint path",
+  metrics: "utilization",
+  examples: [["eth0"], ["enp?s*"]]
+) %>
```

Comment:
Actually, I realized it'd be nice to list out all of the metrics collected for each OS in the docs somewhere.